src/transformers/modeling_utils.py

```diff
@@ -352,11 +352,17 @@ def num_parameters(self, only_trainable: bool = False, exclude_embeddings: bool
             :obj:`int`: The number of parameters.
         """
 
-        def parameter_filter(x):
-            return (x.requires_grad or not only_trainable) and not (isinstance(x, nn.Embedding) and exclude_embeddings)
+        if exclude_embeddings:
+            embedding_param_names = [
+                name + ".weight" for name, module_type in self.named_modules() if isinstance(module_type, nn.Embedding)
```

Comment:
```suggestion
                f"{name}.weight" for name, module_type in self.named_modules() if isinstance(module_type, nn.Embedding)
```
Absolute nit, but I like f-strings.